src/map_cell.cpp

```diff
@@ -0,0 +1,38 @@
+#include <iostream>
+#include <array>
+#include <memory>
+#include <simulator/map_cell.hpp>
+#include <simulator/bug.hpp>
+
+namespace simulation
+{
+
+MapCell::MapCell(const std::shared_ptr<Bug> bug, const AdyacentCellsReference& c_ref) 
+  : _bug(bug), _adyacent_cells{c_ref.up, c_ref.left, c_ref.down, c_ref.right} {}
```

Comment:
Can you add a copy constructor for AdyacentCellsReference so it is easier to copy an existent object?

Reply:
I can, but that won't fix the issue pointed out here. `_adyacent_cells` is actually an `std::array` and not a `AdyacentCellsReference`.  So, that being said, a valid follow up question is

> Why not use the `AdyacentCellsReference` struct directly then?

The answer is that an std::array has already many built in methods and features that would have to be implemented in the newly added struct otherwise. For instance, it's easy to iterate over the elements of an std::array, where there's no way to do so in an `AdyacentCellsReference` unless some new functionality is added/implemented.

Does this make sense?

Reply:
Got it!